tests/queries/0_stateless/02158_groupsortedarray.sql

```diff
@@ -0,0 +1,20 @@
+
+SELECT groupSortedArray(5)(number) from numbers(100);
+
+SELECT groupSortedArrayWeighted(number, number) from numbers(100);
+
+SELECT groupSortedArrayWeighted(100)(number, number) from numbers(1000);
+
+SELECT groupSortedArrayWeighted(100)(number, -number) from numbers(1000);
+
+SELECT groupSortedArrayWeighted(5)(str, number) FROM (SELECT toString(number) as str, number FROM numbers(10));
+
+SELECT groupSortedArray(5)(text) FROM (select toString(number) as text from numbers(10));
+
+SELECT groupSortedArrayWeighted(5)(text, -number) FROM (select toString(number) as text, number from numbers(10));
+
+SELECT groupSortedArray(10)(number, number) from numbers(100); -- { serverError 42 }
+
+SELECT groupSortedArrayWeighted(10)(number) from numbers(100); -- { serverError 42 }
+
+SELECT groupSortedArrayWeighted(5)(number, text) FROM (select toString(number) as text, number from numbers(10)); -- { serverError 43 }
```

Comment:
Consider testing with other types aside from String and Int64. I'd recommend testing things like `Float64`, `Nullable(Float32)`, `Int256`, etc. This is because I fear that the serialization / deserialization isn't correct, so testing with -State and -Merge should bring some peace of mind.

Also, since you have reimplemented addBatch, which covers the -If suffix aggregation, I would test that too.

Lastly, since this is intended to use as a better performance topK (keeping duplicates too) then a performance check is in order to measure things out.